wave_utils/include/wave/utils/world_frame_conversions.hpp

```diff
@@ -0,0 +1,133 @@
+ |                           /    (#UNIVERSITY#)    \                         |
+ |                           |[**](#OFWATERLOO#)[**]|                         |
+ |                           \______________________/                         |
+ |                            |_""__|_,----,_|__""_|                          |
+ |                            ! !                ! !                          |
+ |                            '-'                '-'                          |
+ |       __    _   _  _____  ___  __  _  ___  _    _  ___  ___   ____  ____   |
+ |      /  \  | | | ||_   _|/ _ \|  \| |/ _ \| \  / |/ _ \/ _ \ /     |       |
+ |     / /\ \ | |_| |  | |  ||_||| |\  |||_|||  \/  |||_||||_|| \===\ |====   |
+ |    /_/  \_\|_____|  |_|  \___/|_| \_|\___/|_|\/|_|\___/\___/ ____/ |____   |
+ |                                                                            |
+ ******************************************************************************
+ * ############################################################################
+ *
+ * File: world_frame_conversions.hpp
+ * Desc: Header file for world frame conversion functions
+ * Auth: Michael Smart <michael.smart@uwaterloo.ca>
+ *
+ * ############################################################################
+*/
+
+#ifndef WAVE_SPATIAL_UTILS_WORLD_FRAME_CONVERSIONS_HPP
+#define WAVE_SPATIAL_UTILS_WORLD_FRAME_CONVERSIONS_HPP
+
+#include <cmath>
+#include <GeographicLib/Geocentric.hpp>
+#include <GeographicLib/LocalCartesian.hpp>
+
+namespace wave {
+
+/** Converts a point from LLH (Latitude [deg], Longitude [deg], Height[m]) to
+ *  ECEF.
+ *
+ *  @param[in] llh the input llh point as (Latitude, Longitude, Height). Height
+ *  is relative to the WGS84 ellipsoid.
+ *  @param[out] ecef the corresponding point in the geocentric ECEF frame.
+ */
+void ecefPointFromLLH(const double llh[3], double ecef[3]);
+
+/** Converts a point from LLH (Latitude [deg], Longitude [deg], Height[m]) to
+ *  ECEF.
+ *
+ *  @param[in] ecef the input point in the geocentric ECEF frame.
+ *  @param[out] llh the corresponding llh point as (Latitude, Longitude,
+ *  Height). Height is relative to the WGS84 ellipsoid.
+ */
+void llhPointFromECEF(const double ecef[3], double llh[3]);
+
+/** Computes the 3D Affine transform from ECEF to a local datum-defined ENU
+ *  frame as a 4x4 row-major matrix.
+ *
+ *  @param[in] datum the LLH datum point defining the transform. If
+ *  /p datum_is_LLH is set to false, then the datum values are taken as ECEF
+ *  instead.
+ *  @param[out] T_enu_ecef the 4x4 row-major transformation matrix converting
+ *  column-vector points from ECEF to the local ENU frame defined by the datum
+ *  point.
+ *  @param[in] datum_is_llh \b true: The given datum values are LLH
+ *  (default). <BR>
+ *  \b false: The given datum values are ECEF
+ */
+void enuFromECEFTransformMatrix(const double datum[3],
+                                double T_enu_ecef[4][4],
```

Comment:
Yeah. This guy should be an `Eigen::Affine3d`. Maybe we can get this PR merged in, and then for the next set of changes make these use Eigen objects.

Reply:
will address in #273